tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules2/RewriterScope.scala

```diff
@@ -0,0 +1,11 @@
+package at.forsyte.apalache.tla.bmcmt.rules2
+
+import at.forsyte.apalache.tla.bmcmt.{Binding, PureArena}
+
+/**
+ * A RewriterScope holds data, which is either a byproduct of rewriting, or required supplementary information.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+sealed case class RewriterScope(arena: PureArena, binding: Binding)
```

Comment:
sorry to quibble, but I think this kind of doc is not helpful. `foo holds data` is not an informative thing to say of a struct :)

I think helpful docs for structs tend to explain how a thing is (not just that it is used) and why it holds the data it does. I cannot tell from this what either the arena or the `binding` are used for.